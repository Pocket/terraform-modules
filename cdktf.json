{
  "language": "typescript",
  "app": "npm run --silent compile && node dist/src/example.js",
  "terraformProviders": [
    "pagerduty/pagerduty@~> 1.8.0",
    "archive@~> 1.3.0",
    "local@~> 2.1.0"
  ]
}
