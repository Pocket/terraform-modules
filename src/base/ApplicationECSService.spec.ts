import { Testing, TerraformStack } from 'cdktf';
import {
  ApplicationECSService,
  ApplicationECSServiceProps,
} from './ApplicationECSService';

const BASE_CONFIG: ApplicationECSServiceProps = {
  prefix: 'abides-',
  name: 'ecsiguess',
  ecsCluster: 'gorp',
  vpcId: 'myhouse',
  taskDefinition: {
    port: 8080,
    containerName: 'dudescar',
    containerDefinitions: [
      {
        containerName: 'dudescar',
      },
    ],
  },
};

test('renders an ECS service with minimal config', () => {
  const app = Testing.app();
  const stack = new TerraformStack(app, 'test');

  new ApplicationECSService(stack, 'testECSService', BASE_CONFIG);

  expect(Testing.synth(stack)).toMatchSnapshot();
});

test('renders an ECS service with full container definition props', () => {
  const app = Testing.app();
  const stack = new TerraformStack(app, 'test');

  BASE_CONFIG.taskDefinition.containerDefinitions = [
    {
      containerName: 'dudescar',
      containerImage: 'beverage-here/0.1',
      envVars: {
        rug: 'tiedtheroomtogether',
      },
      secretEnvVars: {
        donnie: 'throwinrockstonight',
      },
    },
  ];

  new ApplicationECSService(stack, 'testECSService', BASE_CONFIG);

  expect(Testing.synth(stack)).toMatchSnapshot();
});

test('renders an ECS service with full container definition props and ALB security group config', () => {
  const app = Testing.app();
  const stack = new TerraformStack(app, 'test');

  BASE_CONFIG.taskDefinition.containerDefinitions = [
    {
      containerName: 'dudescar',
      containerImage: 'beverage-here/0.1',
      envVars: {
        rug: 'tiedtheroomtogether',
      },
      secretEnvVars: {
        donnie: 'throwinrockstonight',
      },
    },
  ];

  BASE_CONFIG.albSecurityGroupConfig = {
    containerPort: 3000,
    albSecurityGroupId: 'strike',
  };

  new ApplicationECSService(stack, 'testECSService', BASE_CONFIG);

  expect(Testing.synth(stack)).toMatchSnapshot();
});
