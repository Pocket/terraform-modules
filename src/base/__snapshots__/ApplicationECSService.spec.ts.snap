// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an ECS service with full container definition props 1`] = `
"{
  \\"//\\": {
    \\"metadata\\": {
      \\"version\\": \\"stubbed\\",
      \\"stackName\\": \\"test\\"
    }
  },
  \\"resource\\": {
    \\"aws_security_group\\": {
      \\"test_testECSService_ecssecuritygroup_AF33BCF2\\": {
        \\"description\\": \\"Internal ECS Security Group\\",
        \\"egress\\": [
          {
            \\"from_port\\": 0,
            \\"protocol\\": \\"-1\\",
            \\"to_port\\": 0,
            \\"cidr_blocks\\": [
              \\"0.0.0.0/0\\"
            ]
          }
        ],
        \\"ingress\\": [],
        \\"name\\": \\"abides--ecsiguess-ECSSecurityGroup\\",
        \\"vpc_id\\": \\"myhouse\\",
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/testECSService/ecs_security_group\\",
            \\"uniqueId\\": \\"test_testECSService_ecssecuritygroup_AF33BCF2\\"
          }
        }
      }
    }
  }
}"
`;

exports[`renders an ECS service with full container definition props and ALB security group config 1`] = `
"{
  \\"//\\": {
    \\"metadata\\": {
      \\"version\\": \\"stubbed\\",
      \\"stackName\\": \\"test\\"
    }
  },
  \\"resource\\": {
    \\"aws_security_group\\": {
      \\"test_testECSService_ecssecuritygroup_AF33BCF2\\": {
        \\"description\\": \\"Internal ECS Security Group\\",
        \\"egress\\": [
          {
            \\"from_port\\": 0,
            \\"protocol\\": \\"-1\\",
            \\"to_port\\": 0,
            \\"cidr_blocks\\": [
              \\"0.0.0.0/0\\"
            ]
          }
        ],
        \\"ingress\\": [
          {
            \\"from_port\\": 80,
            \\"protocol\\": \\"TCP\\",
            \\"to_port\\": 3000,
            \\"security_groups\\": [
              \\"strike\\"
            ]
          }
        ],
        \\"name\\": \\"abides--ecsiguess-ECSSecurityGroup\\",
        \\"vpc_id\\": \\"myhouse\\",
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/testECSService/ecs_security_group\\",
            \\"uniqueId\\": \\"test_testECSService_ecssecuritygroup_AF33BCF2\\"
          }
        }
      }
    }
  }
}"
`;

exports[`renders an ECS service with minimal config 1`] = `
"{
  \\"//\\": {
    \\"metadata\\": {
      \\"version\\": \\"stubbed\\",
      \\"stackName\\": \\"test\\"
    }
  },
  \\"resource\\": {
    \\"aws_security_group\\": {
      \\"test_testECSService_ecssecuritygroup_AF33BCF2\\": {
        \\"description\\": \\"Internal ECS Security Group\\",
        \\"egress\\": [
          {
            \\"from_port\\": 0,
            \\"protocol\\": \\"-1\\",
            \\"to_port\\": 0,
            \\"cidr_blocks\\": [
              \\"0.0.0.0/0\\"
            ]
          }
        ],
        \\"ingress\\": [],
        \\"name\\": \\"abides--ecsiguess-ECSSecurityGroup\\",
        \\"vpc_id\\": \\"myhouse\\",
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/testECSService/ecs_security_group\\",
            \\"uniqueId\\": \\"test_testECSService_ecssecuritygroup_AF33BCF2\\"
          }
        }
      }
    },
    \\"aws_ecr_repository\\": {
      \\"test_testECSService_ecrdudescar_ecrrepo_51BB4E65\\": {
        \\"name\\": \\"abides--ecsiguess-dudescar\\",
        \\"image_scanning_configuration\\": [
          {
            \\"scan_on_push\\": true
          }
        ],
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/testECSService/ecr-dudescar/ecr-repo\\",
            \\"uniqueId\\": \\"test_testECSService_ecrdudescar_ecrrepo_51BB4E65\\"
          }
        }
      }
    },
    \\"aws_ecr_lifecycle_policy\\": {
      \\"test_testECSService_ecrdudescar_ecrrepolifecyclepolicy_0ECE8060\\": {
        \\"policy\\": \\"{\\\\\\"rules\\\\\\":[{\\\\\\"rulePriority\\\\\\":1,\\\\\\"description\\\\\\":\\\\\\"expire old images\\\\\\",\\\\\\"selection\\\\\\":{\\\\\\"tagStatus\\\\\\":\\\\\\"any\\\\\\",\\\\\\"countType\\\\\\":\\\\\\"imageCountMoreThan\\\\\\",\\\\\\"countNumber\\\\\\":800},\\\\\\"action\\\\\\":{\\\\\\"type\\\\\\":\\\\\\"expire\\\\\\"}}]}\\",
        \\"repository\\": \\"abides--ecsiguess-dudescar\\",
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/testECSService/ecr-dudescar/ecr-repo-lifecyclepolicy\\",
            \\"uniqueId\\": \\"test_testECSService_ecrdudescar_ecrrepolifecyclepolicy_0ECE8060\\"
          }
        }
      }
    }
  }
}"
`;
